### DISCUSSION TITLE
422 Unprocessable Entity

### DISCUSSION THREAD

**Discussion by SgNao** [fattureincloud/api#175] [Q&A] [answered]
Buongiorno a tutti,
sto provando a seguire la guida https://developers.fattureincloud.it/docs/guides/invoice-creation/ ma ottengo questo errore:
Exception when calling IssuedDocumentsApi->createIssuedDocument: [422] Client error: `POST https://api-v2.fattureincloud.it/c/{MIO_COMPANY_ID}/issued_documents` resulted in a `422 Unprocessable Entity` response:
{"error":{"message":"Il file specificato non esiste.","validation_result":null}}

Di seguito il codice implementato

```
$entity = new Entity;
$entity
	->setName("Mario Rossi")
	->setVatNumber("47803200154")
	->setTaxCode("RSSMRA91M20B967Q")
	->setAddressStreet("Via Italia, 66")
	->setAddressPostalCode("20900")
	->setAddressCity("Milano")
	->setAddressProvince("MI")
	->setCountry("Italia");

$invoice = new IssuedDocument;
$invoice->setType(IssuedDocumentType::INVOICE);
$invoice->setEntity($entity);

$invoice->setDate(new DateTime("2022-11-28"));
$invoice->setNumber(1);
$invoice->setNumeration("/fatt");
$invoice->setCurrency(
	new Currency(
		array(
				 "id" => "EUR" 
		)
	)
);
$invoice->setItemsList(
	array(
		new IssuedDocumentItemsListItem(
			array(
				"code" => "TV3",
				"name" => "Tavolo in legno",
				"net_price" => 100,
				"category" => "cucina",
				"discount" => 0,
				"qty" => 1,
				"vat" => new VatType(
					array(
						"id" => 0
					)
				)
			)
		)
	)
);
$invoice->setLanguage(
	new Language(
		array(
			"code" => "it",
			"name" => "italiano"
		)
	)
);
$invoice->setEInvoice(true);
$invoice->setEiData(
	new IssuedDocumentEiData(
		array(
			"vat_kind" => "I",
			"payment_method" => "MP05"
		)
	)
);
$invoice->setAttachmentToken({TOKEN});
$invoice->setTemplate(
	new DocumentTemplate(
		array(
			"id" => {TEMPLATE_ID}
		)
	)
);

$create_issued_document_request = new CreateIssuedDocumentRequest;
$create_issued_document_request->setData($invoice);

try {
	$result = $api->createIssuedDocument(FattureInCloud::$companyId, $create_issued_document_request);
	print_r($result);
} catch (Exception $e) {
	echo 'Exception when calling IssuedDocumentsApi->createIssuedDocument: ', $e->getMessage(), PHP_EOL;
}
```

Rimuovendo:
`$invoice->setAttachmentToken({token});`
ottengo
Exception when calling IssuedDocumentsApi->createIssuedDocument: [403] Client error: `POST https://api-v2.fattureincloud.it/c/{MIO_COMPANY_ID}/issued_documents` resulted in a `403 Forbidden` response: {"error":{"message":"No permission.","code":"NO_PERMISSION"}}

Grazie

**Comment by valmoz** [comment, marked as answer]
Buongiorno.
Vedo che lei sta usando il codice d'esempio. Purtroppo quel codice non potrà mai funzionare, ma è solo a scopo illustrativo. Ciò è dovuto al fatto che non possiamo conoscere a priori gli ID delle risorse che fanno parte del vostro specifico account.

Per questo motivo abbiamo preparato la guida seguente, destinata ad aiutarvi a compilare una richiesta di creazione issued document da 0: [Invoice Creation Guide](https://developers.fattureincloud.it/docs/guides/invoice-creation)
le consigliamo caldamente di seguirla passo passo per costruire una richiesta valida.

Per quanto riguarda la sua richiesta:

- setAttachmentToken è un ID che viene associato ad un nuovo attachment al momento del caricamento, per questo restituisce il messaggio "Il file specificato non esiste": il token d'esempio non esiste, oppure esiste ma non è accessibile dal suo account.
- L'errore no permission è relativo alla mancanza di permessi (o scope). La invito a leggere la seguente pagina: [Scopes](https://developers.fattureincloud.it/docs/basics/scopes)

Per aiutarla a iniziare, le suggerisco di iniziare dalla pagina [Getting Started](https://developers.fattureincloud.it/docs/getting-started), dove troverà una spiegazione di tutte le risorse a disposizione.

Grazie
Mauro

**Reply by SgNao** [reply]
Salve, la ringrazio
Gli id sono stati sostituiti con gli id del mio account. 

Seguendo la guida nella fase 8, se effettuo la chiamata ottengo {"error":{"message":"Invalid request.","validation_result":{"filename":["The filename field is required."],"attach.... Ma nella documentazione filename e attachment sono optional

Grazie

**Reply by emajo** [reply]
Ciao @SgNao creando un attachment sono richiesti entrambi i campi, sia filename che attachment, cerchiamo di migliorare sempre la nostra documentazione, grazie per la segnalazione,

Emanuele.
